forest/daemon/src/daemon.rs

```diff
@@ -396,20 +398,43 @@ pub(super) async fn start(
         unblock_parent_process()?;
     }
 
-    // Fetch and ensure verification keys are downloaded
+    // Sets proof parameter file download path early, the files will be checked and
+    // downloaded later right after snapshot import step
     if cns::FETCH_PARAMS {
-        use forest_paramfetch::{
-            get_params_default, set_proofs_parameter_cache_dir_env, SectorSizeOpt,
-        };
-        set_proofs_parameter_cache_dir_env(&config.client.data_dir);
-
-        get_params_default(&config.client.data_dir, SectorSizeOpt::Keys).await?;
+        forest_utils::proofs_api::paramfetch::set_proofs_parameter_cache_dir_env(
+            &config.client.data_dir,
+        );
     }
 
     let config = maybe_fetch_snapshot(should_fetch_snapshot, config).await?;
 
+    let validate_height = if config.client.snapshot {
+        config.client.snapshot_height
+    } else {
+        Some(0)
+    };
+
+    let proof_params_required = !opts.halt_after_import || validate_height.is_some();
+    let ensure_params_downloaded_task_handle = if proof_params_required {
+        Some(tokio::spawn(async {
+            // Add a small delay to avoid import chain progress bar being interrupted.
+            tokio::time::sleep(Duration::from_millis(100)).await;
+            forest_utils::proofs_api::paramfetch::ensure_params_downloaded().await
+        }))
+    } else {
+        None
+    };
```

Comment:
I don't think the performance benefits are worth the extra complexity. Let's just ensure that the parameters have been downloaded right before `validate_chain` and right before `p2p_service` is spawned.

Reply:
Fixed.